ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -75,6 +95,22 @@ private[kvutils] object InputsAndEffects {
       InsertOrdSet.fromSeq(parties.toList.sorted.map(partyStateKey))
     }
 
+    def foldValue(inputs: InsertOrdSet[DamlStateKey], v: Value[ContractId]): InsertOrdSet[DamlStateKey] = {
+      @tailrec def go(inputs: InsertOrdSet[DamlStateKey], vs: Seq[Value[ContractId]]): InsertOrdSet[DamlStateKey] = {
+        vs match {
+          case Nil => inputs
+          case ValueParty(p) :: tail => go(inputs ++ partyInputs(Set(p)), tail)
```

Comment:
`::` is unsafe to use for `Seq` as the backing storage might not be a list, e.g. you would get a runtime no-match exception. We've been hit by this multiple times. Use `+:` or explicitly use `List` instead of `Seq` (this is why you see us not using `Seq` much...)

Reply:
Very good catch! :1st_place_medal: